### DISCUSSION TITLE
Timestamp for FinancialEvents/ServiceFeeEventList

### DISCUSSION THREAD

**Discussion by therubioftw** [amzn/selling-partner-api-models#578] [Ideas]
**Feature request** - **Please add a "postedDate" field to the ServiceFeeEventList events in the listFinancialEvents operation for the Finances API.**

Currently it's the only event type (that I can see) without one, or any unique identifier for that matter, and it makes it VERY difficult to correctly aggregate. Basically there's no way to dedupe correctly, and requires a lot of extra steps on our end.

If anyone here has an easy solution for this, I'll be happy to hear it :)
- Perhaps there's a report that does the aggregation for us? I haven't seen one that covers past orders, historically.
- Perhaps there's a calculation per order that works?

Thank you!

**Comment by takeno-dev** [comment]
I think that is the problem too.
I want to get these Date so I am running the process in periods of 1 day each. It is very annoying.

**Comment by phstc** [comment]
@therubioftw @takeno-dev did any of you find a solution to keep that in sync? I would like to sync it on a daily basis. 

I'm thinking of backtracking and syncing from the last N - 1 day to N days, where N is already in the past. I just need to make sure I don't miss any entry.

Have you also considered using `GET_V2_SETTLEMENT_REPORT_DATA_FLAT_FILE_V2`? It does carry a date.
